// bmi-calculator/Gruntfile.js
module.exports = function (grunt) {
    
    "use strict";

    grunt.initConfig({
        pkg: grunt.file.readJSON('package.json'),
        uglify:{
            dist:{
                options: {
                    banner: '/*! <%= pkg.name %> <%= grunt.template.today("yyyy-mm-dd") %> */\n'
                },
                files:[
                    {"public_html/dist/js/settings.min.js":"public_html/src/js/settings.js"},
                    {"public_html/dist/js/plugin/jquery.bmi.min.js":"public_html/src/js/plugin/jquery.bmi.js"}
                ]
            }
        },
        cssmin: {
            dist: {
                files: [
                    {"public_html/dist/css/default.min.css":"public_html/src/css/default.css"}
                ]
            }
        },
        watch: {
            scripts: {
                files: [
                    'public_html/src/css/*.css',
                    'public_html/src/js/plugin/*.js', 
                    'public_html/src/js/*.js'
                ],
                tasks: ['cssmin', 'uglify'],
                options: {
                    spawn: false
                }
            } 
        }
    });
    
    require('load-grunt-tasks')(grunt, {
        pattern: 'grunt-contrib-*',
        scope: 'devDependencies'
    });
    
    grunt.registerTask('default', [
        'cssmin', 
        'uglify'
    ]);
};
